### PR TITLE
[uss_qualifier/scenarios/utm/data_exchange_validation] Factor away as test step fragment 'get_mock_uss_interactions'

### PR DESCRIPTION
**Pull request by mickmis**
Issue #797
Stacked on top of #831, please only consider last commit.
This factors away inside `interuss/mock_uss` as a test step fragment the retrieval of mock USS interactions.

### CODE REVIEW

File: monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    with scenario.check(
+        "Mock USS interactions logs retrievable", [mock_uss.participant_id]
+    ) as check:
+        try:
+            interactions, query = mock_uss.get_interactions(since)
+            scenario.record_query(query)
+        except QueryError as e:
+            for q in e.queries:
+                scenario.record_query(q)
```
Comment (Shastick) [inline monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py:36]: Nit: scenario has a record_queries() function

File: monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+def notif_op_intent_id_filter(
+    op_intent_id: EntityID,
+) -> Callable[[Interaction], bool]:
+    """Returns an `is_applicable` function that detects whether an op intent notification refers to the specified operational intent."""
+
+    def is_applicable(interaction: Interaction) -> bool:
+        if "json" in interaction.query.request and interaction.query.request.json:
+            return (
+                interaction.query.request.json.get("operational_intent_id", None)
```
Comment (Shastick) [inline monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py:60]: This assumes that any query having operational_intent_id as a first-level field in its body will be a notification, right?

I guess that is fine, but possibly we could also include a call to the operation_filter? Or is the intention that callers/users of the filtering logic would do that?

**Review by Shastick — APPROVED** [review]
Looks good overall, only some minor nits.

**Review by BenjaminPelletier — APPROVED** [review]

File: monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py (hunk 2 above)
Comment (mickmis) [reply monitoring/uss_qualifier/scenarios/interuss/mock_uss/test_steps.py:60]: Yes, and indeed the intention is to let callers deal with that by providing atomic functions.

**Pull request merged**